### PR TITLE
[sfputil] Refactor to utilize sonic-platform package

### PR DESCRIPTION
**Pull request by jleveque**
#### What I did

Refactor sfputil to use sonic-platform package in lieu of old, deprecated platform plugins.

The sfputil utility is still useful, as sfpshow only reads and displays SFP data from State DB. However, we still need a utility to write to the transceivers (low power mode, reset, etc.), and the ability to read directly from the transceivers is helpful for debugging.

#### How I did it

- Complete refactor to use sonic-platform package
- Refactor display functions to align with sfpshow utility
- Bump version to 3.0
- Add unit tests

#### How to verify it

Run all `sfputil` commands, ensure they work. Do the same on multiple platforms. Output should be the same as before.

### CODE REVIEW

File: sfputil/main.py [hunk 1]
```diff
@@ -5,118 +5,386 @@
...
+import os
+import sys
 
-    import click
-    from sonic_py_common import device_info, logger, multi_asic
-    from tabulate import tabulate
-except ImportError as e:
-    raise ImportError("%s - required module not found" % str(e))
+import click
+import sonic_platform
```
Comment (sanmalho-git) [inline sfputil/main.py:13]: @jleveque - We tested this file on a multi-asic and single asic box, except we had to change this line to
<pre>
import sonic_platform.platform
</pre>

In common code, we are seeing a lot of places where we have to make the same change as above for python3 to work. Is this the best way to fix this?

Comment (jleveque) [reply sfputil/main.py:13]: I have heard other accounts of this. Why do you need to change it? What is the error you encounter with it like this?

Comment (sanmalho-git) [reply sfputil/main.py:13]: admin@Linecard2:$ /usr/bin/python3
Python 3.7.3 (default, Jul 25 2020, 13:03:44) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sonic_platform
>>> platform_chassis = sonic_platform.platform.Platform().get_chassis()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'sonic_platform' has no attribute 'platform'
>>>

Comment (jleveque) [reply sfputil/main.py:12]: I don't see this issue on Arista, Dell or Mellanox platforms. Can you see if there is a noticeable difference in the way your sonic-platform packages are structured?

Also, I just pushed a couple new commits. Can you please test the latest changes?

**Comment by jleveque** [comment]
Retest this please

File: sfputil/main.py (hunk 1 above)
Comment (sanmalho-git) [reply sfputil/main.py:12]: @jleveque sorry for the delay.  There was an issue in the packaging of our sonic-platform that we have now resolved and thus don't see the backtrace with the original contents.

Also, I have tested the latest main.py on multi-asic and single asic pizza box, and it works fine.

**Comment by mlok-nokia** [comment]
Should we also change psuutil as well to use the new PMON 2.0 API's?

**Comment by jleveque** [comment]
> Should we also change psuutil as well to use the new PMON 2.0 API's?

I just merged those changes, actually: https://github.com/Azure/sonic-utilities/pull/1434

File: sfputil/main.py (hunk 1 above)
Comment (jleveque) [reply sfputil/main.py:12]: Great to hear! And thanks for testing the changes!

**Review by lguohan — APPROVED** [review]

**Pull request merged**